pkg/scalers/azure/azure_eventhub.go

```diff
@@ -19,15 +22,20 @@ type EventHubInfo struct {
 	Namespace                string
 	EventHubName             string
 	CheckpointStrategy       string
-	Cloud                    string
 	ServiceBusEndpointSuffix string
 	ActiveDirectoryEndpoint  string
+	EventHubResourceURL      string
+	PodIdentity              kedav1alpha1.PodIdentityProvider
 }
 
+const (
+	DefaultEventhubResourceURL = "https://eventhubs.azure.net/"
+)
+
 // GetEventHubClient returns eventhub client
-func GetEventHubClient(info EventHubInfo) (*eventhub.Hub, error) {
+func GetEventHubClient(ctx context.Context, info EventHubInfo) (*eventhub.Hub, error) {
```

Comment:
Shouldn't we align with the other scalers and use a switch instead?

Reply:
Could you elaborate a bit more on this? Align what with the other scalers?

Reply:
The other scalers use a switch case and here we use an if.

Reply:
Oh, got it. GitHub highlighted line 36 with the function definition, so I was wondering if you meant something needed to be changed there. Yeah, will change it to a switch.

Reply:
Done.